src/generic/bmpsvg.cpp

```diff
@@ -28,14 +55,173 @@
+    wxBitmapBundleImplSVG(std::unique_ptr<wxlunasvg::Document> svg_document, const wxSize& sizeDef);
+
+    virtual wxSize GetDefaultSize() const override;
+    virtual wxSize GetPreferredBitmapSizeAtScale(double scale) const override;
+
+    virtual wxBitmap GetBitmap(const wxSize& size) override;
+
+    bool IsOk() const;
+
+private:
+    wxBitmap DoRasterize(const wxSize& size);
+
+    std::unique_ptr<wxlunasvg::Document> m_svgDocument;
+
+    const wxSize m_sizeDef;
+
+    // Cache the last used bitmap (may be invalid if not used yet).
+    //
+    // Note that we cache only the last bitmap and not all the bitmaps ever
+    // requested from GetBitmap() for the different sizes because there would be
+    // no way to clear such cache and its growth could be unbounded, resulting
+    // in too many bitmap objects being used in an application using SVG for all
+    // of its icons.
+    wxBitmap m_cachedBitmap;
+
+    wxDECLARE_NO_COPY_CLASS(wxBitmapBundleImplSVG);
+};
+
+// ============================================================================
+// wxBitmapBundleImplSVG implementation
+// ============================================================================
+
+wxBitmapBundleImplSVG::wxBitmapBundleImplSVG(std::unique_ptr<wxlunasvg::Document> svg_document, const wxSize& sizeDef)
+    : m_sizeDef(sizeDef)
+{
+    m_svgDocument = std::move(svg_document);
+}
+
+/* static */
+wxBitmapBundle wxBitmapBundle::FromSVG(const char* data, const wxSize& sizeDef)
+{
+    wxCHECK_MSG(data != nullptr, wxBitmapBundle(), "null data pointer");
+    wxCHECK_MSG(sizeDef.GetWidth() > 0 && sizeDef.GetHeight() > 0, wxBitmapBundle(), "invalid default size");
+    auto svgDocument = wxlunasvg::Document::loadFromData(data);
+    return wxBitmapBundle(new wxBitmapBundleImplSVG(std::move(svgDocument), sizeDef));
+}
+
+/* static */
+wxBitmapBundle wxBitmapBundle::FromSVG(char* data, const wxSize& sizeDef)
+{
+    wxCHECK_MSG(data != nullptr, wxBitmapBundle(), "null data pointer");
+    wxCHECK_MSG(sizeDef.GetWidth() > 0 && sizeDef.GetHeight() > 0, wxBitmapBundle(), "invalid default size");
+    auto svgDocument = wxlunasvg::Document::loadFromData(data);
+    return wxBitmapBundle(new wxBitmapBundleImplSVG(std::move(svgDocument), sizeDef));
+}
+
+/* static */
+wxBitmapBundle wxBitmapBundle::FromSVG(const wxByte* data, size_t len, const wxSize& sizeDef)
+{
+    wxCHECK_MSG(data != nullptr, wxBitmapBundle(), "null data pointer");
+    wxCHECK_MSG(len > 0, wxBitmapBundle(), "zero length data");
+    wxCHECK_MSG(sizeDef.GetWidth() > 0 && sizeDef.GetHeight() > 0, wxBitmapBundle(), "invalid default size");
+    auto svgDocument = wxlunasvg::Document::loadFromData(reinterpret_cast<const char*>(data), len);
+    return wxBitmapBundle(new wxBitmapBundleImplSVG(std::move(svgDocument), sizeDef));
+}
+
+bool wxBitmapBundleImplSVG::IsOk() const
+{
+    return m_svgDocument != nullptr;
+}
+
+wxBitmap wxBitmapBundleImplSVG::DoRasterize(const wxSize& size)
+{
+    if (IsOk())
```

Comment:
```suggestion
    if ( IsOk() )
```

Per coding guidelines, the condition in parentheses should be "wrapped" in a space, applies to other cases here.

Also, I personally would just return here, instead of adding a huge indented block below.